### DISCUSSION TITLE
preprocess and postprocess

### DISCUSSION THREAD

**Discussion by gitlabspy** [tensorflow/compression#146] [Q&A] [answered]
Let's say we have an arbitrary size image, it should be preprocessed before feeding into the compression model. When decompression, we retrieve the bitstream and  reconstruct the image, then we do postprocess to retrieve the image with its original size.
If we decided the model input shape as `[128, 128]` and we have two images with shape `[256, 128]` and `[64, 64]`, how will they be preprocessed and postprocessed?

**Comment by jonaballe** [comment, marked as answer]
All the pre-trained models accessible with `tfci.py` and the model code under `models/` are fully convolutional. So the decoder will always output an image of the original size.
